go/pkg/client/cas.go

```diff
@@ -36,6 +36,33 @@ const DefaultCompressedBytestreamThreshold = -1
 // DefaultMaxHeaderSize is the defaut maximum gRPC header size.
 const DefaultMaxHeaderSize = 8 * 1024
 
+// MovedBytesMetadata represents the bytes moved in CAS related requests.
+type MovedBytesMetadata struct {
+	// Requested is the sum of the sizes in bytes for all the uncompressed
+	// blobs needed by the execution. It includes bytes that might have
+	// been deduped and thus not passed through the wire.
+	Requested int64
+	// LogicalMoved is the sum of the sizes in bytes for all uncompressed
+	// blobs passed through the wire. It may be different
+	LogicalMoved int64
+	// RealMoved is the the sum of sizes in bytes for all blobs passed
+	// through the wire in the format they were passed through (eg
+	// compressed).
+	RealMoved int64
+	// Cached is amount of logical bytes that we did not have to move
+	// through the wire because they were de-duped.
+	Cached int64
+}
+
+func sumStats(a, b MovedBytesMetadata) MovedBytesMetadata {
```

Comment:
nit: we agreed to pass all structs by pointer with the sole exception of Digest. Ditto everywhere else. Same for returning.

Reply:
Done